Source/Blazorise/Utilities/BaseDateTimeFormatConverter.cs

```diff
@@ -0,0 +1,85 @@
+﻿#region Using directives
+using System.Collections.Generic;
+using System.Text;
+#endregion
+
+namespace Blazorise.Utilities;
+
+/// <summary>
+/// Base class for internal date time formaters.
+/// </summary>
+public abstract class BaseDateTimeFormatConverter
+{
+    #region Methods
+
+    /// <inheritdoc/>
+    public string Convert( string format )
+    {
+        if ( string.IsNullOrEmpty( format ) || format.Length == 0 )
+            return format;
+
+        var tokens = ParseTokens( format );
+
+        var endFormat = new StringBuilder();
+
+        foreach ( var token in tokens )
+        {
+            if ( FormatTokens.TryGetValue( token, out var internalToken ) )
+                endFormat.Append( internalToken );
+            else
+                endFormat.Append( token );
+        }
+
+        return endFormat.ToString();
+    }
+
+    private IList<string> ParseTokens( string format )
+    {
+        var tokens = new List<string>();
+
+        string tempToken = null;
+        char? last = null;
+
+        foreach ( var c in format )
+        {
+            if ( !char.IsLetter( c ) )
```

Comment:
What is this doing exactly? Looking for the dashes? '**-**'? Maybe a string.Split is easier to read? And is token a good naming? Seems confusing to me, maybe format templates? or format strings?

Reply:
Or maybe even, more close to what they are, Date Formats, or DateTime Formats?

Reply:
Anything that is not a letter is considered a separator. Like `.`, `/`, `-`, etc.

Reply:
Well even numbers... no? :)

Reply:
That would be an invalid character, but if user wants that who cares 🤷‍♂️🤣